tests/helpers_tests/test_condition.py

```diff
@@ -17,117 +17,342 @@
+      condition_list, {'weird_condition': 'hi'}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exists__returns_false__when_no_user_provided_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exists__returns_false__when_user_provided_value_is_null(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': None}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_string(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 'hi'}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_number(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 10}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 10.0}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_boolean(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': False}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exact_string__returns_true__when_user_provided_value_is_equal_to_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': 'Lacerta'}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exact_string__returns_false__when_user_provided_value_is_not_equal_to_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': 'The Big Dipper'}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exact_string__returns_null__when_user_provided_value_is_different_type_from_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': False}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exact_string__returns_null__when_no_user_provided_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exact_number__returns_true__when_user_provided_value_is_equal_to_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_number_condition_list, {'lasers_count': 9000}
```

Comment:
Can we also find a way to test with `9000L` in the case of Python 2?

Reply:
Actually...*do* we run our unit tests in both Python 2 and Python 3?? @aliabbasrizvi